app/dts/bindings/animations/animation_base.yaml

```diff
@@ -0,0 +1,23 @@
+# Copyright (c) 2020, The ZMK Contributors
+# SPDX-License-Identifier: MIT
+
+properties:
+  pixels:
+    type: array
+    required: true
+    description: |
+      Pixel positions to which the animation should apply.
+
+  blending-mode:
```

Comment:
Just a suggestion here- `blending-mode` could be encoded as string enum, and we could use [DT_ENUM_IDX](https://docs.zephyrproject.org/latest/build/dts/api/api.html#c.DT_ENUM_IDX) to get the current blending mode for an animation.

See here for an example of using string enums:
https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/pinctrl/nxp%2Ckinetis-pinctrl.yaml#L69

Reply:
I guess this is okay if it's optional, but I wonder whether this is really needed in *every* animation. Could the blend mode be a separate yaml file that individual animations can include as needed?

I would think there are many potential animations that would not use a blend mode, or where only one blend mode makes any sense. Examples I can think of:
- An animation that passes straight through to another animation and doesn't draw anything on its own
- An animation that overwrites the colors of certain pixels, e.g. a caps lock indicator
- An animation that only modifies what another animation drew, e.g. something that smoothly fades everything to black when the keyboard goes idle

Reply:
TIL - the string enums look pretty neat.

As for whether this is necessary... not really. I mostly thought it'd be fun to experiment with and most people will probably just leave it at the default most of the time.  
But to give a practical example, let's take a look at the ripple animation. As annoying as it is visually, it's find it's really good to illustrate such things.

First of all, the blending modes setting allows you to run it on top of another animation. So you can have some light show going on while idle'ing, and ripples (or other effects) appear on top of that when typing.  
The expanding circle is in fact a gradient, and due to the nature of a keyboard with pixels being far apart and unevenly distributed, for that to work you need to blend the ripple pixel value with the underlying animation. That's where the blending mode comes in.

TL;DR - While not absolutely critical, I believe it allows for some really cool things down the line and would potentially be a major differentiator between ZMK and other keyboard firmwares which as far as I'm aware, are pretty restrictive in what you can and can't combine and when.

Reply:
I agree that we should at the very least keep blending support as an optional feature- it allows ripple overlays like @ice9js described, and it also allows animation zones to overlap (so you could have static red and blue animations fade into purple in the middle, without needing to create custom colors to calculate that gradient)

Reply:
I was originally thinking that this would be a property used by an animation to decide how to blend two or more child animations, in which case it only makes sense to include on animations that do blend together animations.

If it's instead used to tell a *parent* animation how it should blend this animation such that you could do something like

* composite
  * base: overwrite
  * ripple: add
  * indicators: overwrite

then having this property on every animation does make sense to me.